src/components/Comments.jsx

```diff
@@ -0,0 +1,48 @@
+import { useContext, useEffect, useState } from "react";
+import { UserContext, UserProvider } from "../contexts/UserContext";
+import { getAllArticles, getCommentsByArticleID } from "../utils/api";
+import { Link } from "react-router-dom";
+import { useParams } from "react-router-dom";
+
+function Comments() {
+  const [comments, setComments] = useState("");
+
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(false);
+  const { user, setUser } = useContext(UserContext);
+  const { article_id } = useParams();
+
+  useEffect(() => {
+    getCommentsByArticleID(article_id)
+      .then((data) => {
+        setComments(data);
+        setLoading(false);
+      })
+      .catch((error) => {
+        console.log(error);
+        setError(true);
+      });
+  }, []);
+
+  if (error) {
+    return <p>error!</p>;
+  } else if (loading) {
+    return <p>Loading comments for this article! Please wait a moment.</p>;
```

Comment:
Nice user feedback here in terms of your conditional rendering for both error and loading states.